### DISCUSSION TITLE
Compiler scope by URL / route

### DISCUSSION THREAD

**Discussion by thinkwelltwd** [unpoly/unpoly#351] [Q&A] [answered]
I have a project that uses datatables heavily where the html table id is deterministically generated from the model name when the html page is rendered. Unpoly compilers initialize the `datatables` on the html table so that datatables handling is initialized via full page loads, `up-follow` attributes, modal layers and so forth.

While html table id is generated for a given model, the html table structure can vary depending on which page they're on, so initializing compilers like so

`up.compiler('#databable_model_name', function(element, pins) {`

would cause problems if the compilers are loaded up front and various html tables display the data in different forms throughout the app. If I was always doing entire page loads, then I'd just use a script tag in the head section of a given page and load only the compilers needed for that page.

How do other unpoly devs handle situations where unpoly compilers should be "scoped" by route to fire the correct functions?

**Comment by triskweline** [comment, marked as answer]
Most modern frameworks want you to think in components, not pages. The same component will be rendered the same on any URL. Any variation is pushed to the component's parameters, e.g. props in React or [`[up-data]`](https://unpoly.com/up-data) in Unpoly.

If, in your example, the table is sometimes sorted by user name and sometimes by creation time, the sorting criterion can be a parameter.

**Reply by thinkwelltwd** [reply]
Thanks; I'm following you; I'll have to adjust my thinking. Didn't want to be missing something obvious.
